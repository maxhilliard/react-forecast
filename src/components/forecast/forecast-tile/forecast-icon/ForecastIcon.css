.forecastIcon {
    composes: forecastIcon from '../ForecastTile.css';
    display: flex;
    justify-content: center;
    align-items: center;
}