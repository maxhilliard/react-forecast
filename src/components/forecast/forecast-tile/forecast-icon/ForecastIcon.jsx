import React from 'react';
import PropTypes from 'prop-types';

import styles from './ForecastIcon.css';

const constructIconUrl = iconCode => `http://openweathermap.org/img/w/${iconCode}.png`;

const ForecastIcon = ({ icon, description }) => {
    const iconUrl = constructIconUrl(icon);

    return (
        <div className={styles.forecastIcon}>
            <img src={iconUrl} alt={description} />
        </div>
    );
};

ForecastIcon.propTypes = {
    icon: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
};

export default ForecastIcon;
