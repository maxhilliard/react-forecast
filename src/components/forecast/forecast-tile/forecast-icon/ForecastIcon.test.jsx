import React from 'react';
import { shallow } from 'enzyme';

import ForecastIcon from './ForecastIcon';

describe('ForecastIcon', () => {
    describe('snapshot', () => {
        it('matches snapshot', () => {
            const mockProps = {
                weather: [{
                    icon: 'mock-icon-code',
                    description: 'mock-weather-description',
                }],
            };

            const wrapper = shallow(<ForecastIcon {...mockProps} />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
