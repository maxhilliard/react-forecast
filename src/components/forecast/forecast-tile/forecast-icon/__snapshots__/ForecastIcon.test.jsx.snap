// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastIcon snapshot matches snapshot 1`] = `
<div
  className="forecastIcon"
>
  <img
    src="http://openweathermap.org/img/w/undefined.png"
  />
</div>
`;
